scipy/stats/tests/test_distributions.py

```diff
@@ -4298,6 +4307,17 @@ def test_fit_stats_gh18274(self):
         stats.betaprime.fit([0.1, 0.25, 0.3, 1.2, 1.6], floc=0, fscale=1)
         stats.betaprime(a=1, b=1).stats('mvsk')
 
+    def test_moment_gh18634(self):
+        # Testing for gh-18634 revealed that `betaprime` raised a
+        # NotImplementedError for higher moments. Check that this is
+        # resolved. Parameters are arbitrary but lie on either side of the
+        # moment order (5) to test both branches of `_lazywhere`. Reference
+        # values produced with Mathematica, e.g.
+        # `Moment[BetaPrimeDistribution[2,7],5]`
+        ref = [np.inf, 0.867096912929055]
+        res = stats.betaprime(2, [4.2, 7.1]).moment(5)
+        assert_allclose(res, ref)
```

Comment:
![image](https://github.com/scipy/scipy/assets/6570539/9bdcda11-6ca6-48e9-a7ca-95581cb360b8)

When `b` is less than the moment order, the moment is infinite.